### ISSUE TITLE
cli.py should check_schema before using it.

### ISSUE THREAD

**Issue by Parakleta** [Julian/jsonschema#208]
The documentation suggests "first verify that the provided schema is itself valid, since not doing so can lead to less obvious error messages and fail in less obvious or consistent ways" but the command line utility does not follow this advice.

If it is desirable to not recheck the schema on every run for performance reasons related to scripting it would be possible to instead check the schema just when the cli tool is called with no instances (which currently appears to be effectively a no-op), but I would prefer it on every run (since multiple instances can be provided in a single execution).

**Comment by Julian** [comment]
Yup, agreed. Any chance you'd be interested in a PR?

**Comment by Parakleta** [comment]
Not really.  There are several different ways the issue could be resolved and I don't think it's my place to determine which of those methods is correct, especially since I have no particular understanding of this project, or more than a passing familiarity with Python.

For my purpose because I am calling the validator from other code as well I have changed the `Validator.__init__` function, adding the line `schema is not self.META_SCHEMA and self.check_schema(schema)`.  I chose this because it is more convenient for me to know that any validator object has a legal schema attached, but I don't know how this fits into the scope of your project.

Going forward I think I am likely to abandon JSON schema in general anyway since I need more expressive power, meaning either XSD1.1 or a Lua DSL.
